tsl/test/sql/compression_trigger.sql

```diff
@@ -0,0 +1,95 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+-- This is copied from hypercore_trigger.sql
+
+create table readings(
+       metric_id serial,
+       created_at timestamptz not null unique,
+       location_id smallint,	--segmentby attribute with index
+       owner_id bigint,		--segmentby attribute without index
+       device_id bigint,	--non-segmentby attribute
+       temp float8,
+       humidity float4
+);
+
+select create_hypertable('readings', by_range('created_at'));
+
+select setseed(1);
+
+insert into readings(created_at, location_id, device_id, owner_id, temp, humidity)
+select t, ceil(random()*10), ceil(random()*30), ceil(random() * 5), random()*40, random()*100
+from generate_series('2022-06-01'::timestamptz, '2022-07-01', '5m') t;
+
+alter table readings set (
+	  timescaledb.compress,
+	  timescaledb.compress_orderby = 'created_at',
+	  timescaledb.compress_segmentby = 'location_id, owner_id'
+);
+
+select compress_chunk(show_chunks('readings'));
+
+create table saved_rows (like readings, new_row bool not null, kind text);
+
+create function save_transition_table() returns trigger as $$
+begin
+   case tg_op
+   	when 'INSERT' then
+	     insert into saved_rows select n.*, true, tg_op from new_table n;
+	when 'DELETE' then
+	     insert into saved_rows select o.*, false, tg_op from old_table o;
+	when 'UPDATE' then
+	     insert into saved_rows select n.*, true, tg_op from new_table n;
+	     insert into saved_rows select o.*, false, tg_op from old_table o;
+   end case;
+   return null;
+end;
+$$ language plpgsql;
+
+create trigger save_insert_transition_table_trg
+       after insert on readings
+       referencing new table as new_table
+       for each statement execute function save_transition_table();
+
+insert into readings(created_at, location_id, device_id, owner_id, temp, humidity)
+values ('2022-06-01 00:01:23', 999, 666, 111, 3.14, 3.14),
+       ('2022-06-01 00:02:23', 999, 666, 112, 3.14, 3.14);
+
+select * from saved_rows;
+
+truncate saved_rows;
+
+copy readings(created_at, location_id, device_id, owner_id, temp, humidity) from stdin with (format csv);
+"2022-06-01 00:01:35",999,666,111,3.14,3.14
+\.
+
+select * from saved_rows;
+
+truncate saved_rows;
+
+create trigger save_update_transition_table_trg
+       after update on readings
+       referencing new table as new_table old table as old_table
+       for each statement execute function save_transition_table();
+
+select * from readings where location_id = 999;
+
+update readings set humidity = 99.99 where location_id = 999;
```

Comment:
Would be good to add an example updating and deleting compressed data. WDYT?

Reply:
Fixed this by compressing the table between the operations and it turned out that we cannot support DELETE triggers with transition tables for normal compression since it does not build a transition table when deleting an entire compressed segment.

Added an error for this case.

Reply:
Hmmm... if we create trigger on DELETE before turn ON the compression then we have a problem, no?

Reply:
Yes, it is the same as if we add an index before we turn on compression. I'll take a look and see if we can easily add an error for this when altering the table.

Reply:
Added an error message when you try to set compression options but have a DELETE trigger with a transition table.

Reply:
I didn't see a test case for it... did u added? Or am I missing something?

Reply:
Added.